lib/realtime/broadcast_changes/handler.ex

```diff
@@ -0,0 +1,204 @@
+defmodule Realtime.BroadcastChanges.Handler do
+  use GenServer
+  require Logger
+
+  import Realtime.Adapters.Postgres.Protocol
+  import Realtime.Adapters.Postgres.Decoder
+  import Realtime.Helpers, only: [log_error: 2]
+
+  alias Realtime.Adapters.Postgres.Decoder
+  alias Realtime.Adapters.Postgres.Protocol.KeepAlive
+  alias Realtime.Adapters.Postgres.Protocol.Write
+  alias Realtime.Api.Tenant
+  alias Realtime.BroadcastChanges.PostgresReplication
+  alias Realtime.Database
+  alias Realtime.Tenants.BatchBroadcast
+  alias Realtime.Tenants.Cache
+
+  defstruct [:tenant_id, relations: %{}, buffer: [], postgres_replication_pid: nil]
+
+  @behaviour PostgresReplication.Handler
+  @registry Realtime.BroadcastChanges.Handler.Registry
+
+  @spec name(Tenant.t()) :: term()
+  def name(%Tenant{external_id: tenant_id}) do
+    {:via, Registry, {@registry, tenant_id}}
+  end
+
+  @spec supervisor_spec(Tenant.t()) :: term()
+  def supervisor_spec(%Tenant{external_id: tenant_id}) do
+    {:via, PartitionSupervisor, {Realtime.BroadcastChanges.Handler.DynamicSupervisor, tenant_id}}
+  end
+
+  @impl true
+  def call(message, metadata) when is_write(message) do
+    %{tenant_id: tenant_id} = metadata
+    %Write{message: message} = parse(message)
+
+    case Registry.lookup(@registry, tenant_id) do
+      [{pid, _}] ->
+        message |> decode_message() |> then(&send(pid, &1))
+        :noreply
+
+      _ ->
+        Logger.error("Unable to find BroadcastChanges for tenant: #{tenant_id}")
+        :shutdown
+    end
+  end
+
+  def call(message, _metadata) when is_keep_alive(message) do
+    %KeepAlive{reply: reply, wal_end: wal_end} = parse(message)
+    wal_end = wal_end + 1
+
+    message =
+      case reply do
+        :now -> standby_status(wal_end, wal_end, wal_end, reply)
+        :later -> hold()
+      end
+
+    {:reply, message}
+  end
+
+  def call(msg, state) do
+    Logger.info("Unknown message received: #{inspect(%{msg: parse(msg), state: state})}")
```

Comment:
nitpick: maybe consider warning so it shows up in prod logs? Seems like a bug that this would get unrelated message?

```suggestion
    Logger.warning("Unknown message received: #{inspect(%{msg: parse(msg), state: state})}")
```